src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/impl/SingleNodeLabelMappingToLiteralsImpl.java

```diff
@@ -0,0 +1,39 @@
+package se.liu.ida.hefquin.engine.wrappers.lpgwrapper.impl;
+
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.NodeFactory;
+import org.apache.jena.vocabulary.XSD;
+import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.impl.exceptions.UnSupportedNodeLabelException;
+
+public class SingleNodeLabelMappingToLiteralsImpl implements NodeLabelMapping {
+
+    protected final String label;
+    protected final Node node;
+
+    public SingleNodeLabelMappingToLiteralsImpl(final String label, final String literal){
+        this.label=label;
+        this.node = NodeFactory.createLiteral(literal);
+    }
+
+    @Override
+    public Node map(final String label) {
+        if (label.equals(this.label)) {
+            return this.node;
+        }
+        else {
+            throw new UnSupportedNodeLabelException("The given node label (" + label + ") is not a supported label in the image of this node label mapping.");
+        }
+    }
+
+    public String unmap(final Node node) {
+        if (!isPossibleResult(node))
+            throw new UnSupportedNodeLabelException("The given RDF term (" + node.toString() + ") is not a literal.");
+        return this.label;
+    }
+
+    @Override
+    public boolean isPossibleResult(final Node node) {
+        return node.isLiteral() && XSD.xstring.getURI().equals(node.getLiteral().getDatatypeURI()) && node.getLiteral().toString().equals(this.node.getLiteral().toString());
```

Comment:
This one can be simplified:
```suggestion
        return node.equals( this.node );
```